### PR TITLE
ICEFinder container

### PR DESCRIPTION
**Pull request by Ales-ibt**

### CODE REVIEW

File: src/tools/ICEfinder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+ENV VERSION="1.0"
+
+LABEL maintainer="Microbiome Informatics Team www.ebi.ac.uk/metagenomics"
+LABEL software="ICE_finder"
+LABEL version="${VERSION}"
+LABEL description="A tool for the detection of ICEs/IMEs of bacterial genomes"
+LABEL website="https://bioinfo-mml.sjtu.edu.cn/ICEfinder/ICEfinder.html"
+LABEL license="The Perl-based local version of ICEfinder is available by contact Dr Hong-Yu (hyou@sjtu.edu.cn)"
```
Comment (mberacochea) [inline src/tools/ICEfinder/Dockerfile:12]: The license is:
```
Term

Both licensee and the Microbial Bioinformatics Group, School of Life Sciences & Biotechnology, Shanghai Jiao Tong University (SJTU), agree as follows:

1.The software ICEfinder 1.0 is only available to academic, government and other nonprofit institutions for noncommercial and nonprofit research purposes.

2.Licensee is granted to install and use the Software ICEfinder 1.0 in computers located at the licensee's institution. The software ICEfinder 1.0 may not be transferred or distributed to any other party by Licensee without the prior written approval of Dr. Hong-Yu Ou, SJTU.

3.Title, ownership rights, and intellectual property rights in and to the Software ICEfinder 1.0 shall remain in SJTU. The software ICEfinder 1.0 is protected by copyright laws of the People's Republic of China, and under international treaties.

4.SJTU may terminate this license by writing notification to Licensee.

5.Licensee agrees to cite the following paper, based completely or in part on the use of the Software SJTU, in any publication, presentation or other work:
M. Liu, X. Li, Y. Xie, D. Bi, J. Sun, J. Li, C. Tai, Z. Deng, H.Y. Ou (2019) ICEberg 2.0: an updated database of bacterial integrative and conjugative elements. Nucleic Acids Research, DOI: 10.1093/nar/gky1123.

6.Warranty, Liability and Indemnity. Licensee acknowledges that the Software ICEfinder 1.0 is experimental. The Software ICEfinder 1.0 is provided based on an "as is", " as available" basis, and SJTU makes no representation or warranty, express or implied, including without limitation any warranties of merchantability or fitness for a particular purpose or non-infringement. Neither party shall be liable for any claim, demand or action for any loss, harm, illness or other damage or injury arising from access to or use of the Software ICEfinder 1.0, including without limitation any direct, indirect, incidental, exemplary, special or consequential damages, even if advised of the possibility of such damages.
```
from LICENSE_ICEfinder1.txt.

This is a bit tricky, we need to
- Talk to Rob about this license, I think it's OK
- But, we can't make the container public as the license doesn't allow us to do so.
  - Which is a problem in terms of having this workflow for public usage.
- Email the author as "MGnify" and make our case, maybe we can get en extension of the library or the author allows us to distribute the docker as it is.

File: src/tools/ICEfinder/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# Set environment
+RUN export PERL5LIB="/opt/miniconda/lib/perl5/site_perl/5.22.0/Bio"
+
+## Getting ICE_finder
+# The local version of ICEfinder is located in a googleDrive folder
+# This is a temporary solution to get the tool available
+# Downloading the tar file
+RUN mkdir /install /input /output /work
+WORKDIR /install
+RUN gdown --id 1RlkjErOg08W-MUrO7cLUm_G3FLB2Yq7g
```
Comment (mberacochea) [inline src/tools/ICEfinder/Dockerfile:49]: A better approach is to build the container locally and have the software there, please remove the drive link as it is public (even though no one has it)

**Review by mberacochea — CHANGES_REQUESTED** [review]
The Dockerfiles look good to me :), but we need to chat about ICEfinder.

**Comment by Ales-ibt** [comment]
ICEfinder container will be generated by the user following the instructions on README file after getting their own copy of the tool. Therefore, we don't need this dockerfile anymore

**Pull request closed without merging**